lib/dialog-renderer/index.tsx

```diff
@@ -52,6 +53,8 @@ export class DialogRenderer extends Component<Props> {
               <AboutDialog key="about" closeDialog={closeDialog} />
             ) : 'BETA-WARNING' === dialog ? (
               <BetaWarning key="beta-warning" />
+            ) : 'CLOSE-WINDOW-CONFIRMATION' === dialog ? (
+              <CloseWindowConfirmation key="close-window-confirmation" />
```

Comment:
While I love that the specifics are separated from the dialog I feel like we don't really need to have separate components for them. In this case it almost masquerades the intrinsics of what we're opening. That is, is `<CloseWindowConfirmation />` all that better than this?

```ts
<UnsynchronizedConfirmation
     actionDescription="Closing the app with unsynchronized notes could cause data loss."
     actionName="Close window"
     actionSafeName="Safely close window"
     continueAction={this.props.reallyCloseWindow}
/>
```

This second example appears to break the pattern in this dialog-renderer list but it also means we don't have to go hunting to realize that the close and logout dialogs are basically the same. It also means we don't have to create a seemingly unnecessary `continueAction` prop just for the `<UnsynchronizedConfirmation />`.

Reply:
I disagree. I think adding the complexities of the redux actions and the copy to DialogRenderer is a mistake.

I think if we were to do this it would make more sense to push it back to the component.

Reply:
>I think if we were to do this it would make more sense to push it back to the component.

Can you identify the demonstratives here?

If we were to do what?
Push what back to the component?

Reply:
If we were to remove the separate components and put the props in `lib/dialog-renderer/index.tsx` as you described. This component should not be concerned with the specifics of the logout and close window dialogs.